### DISCUSSION TITLE
Is that possible to use meep to calculate PCB stackup impedance?

### DISCUSSION THREAD

**Discussion by ganghuang** [NanoComp/meep#2586] [Q&A]
The first step for any RF PCB design is determine the stack up and calculate the trace/gap width for typical impedance, e.g. 50 ohm, 100 ohm. 
May I use the meep to verify the calculation like below?
I tried the atlc (https://atlc.sourceforge.net/) and got similar result for single ended co-planar wave guide (50 vs 51 ohm), but for the differential pair, I got ~108.5 ohm instead of 100 ohm as suggested by the vendor. 
I want to try some other EM solver to see which one can do better on this kind of job.
![image](https://github.com/NanoComp/meep/assets/7526459/74da1d41-2210-445d-8529-f4a182ab2f9c)

**Comment by smartalecH** [comment]
You typically want a _mode solver_ for these kinds of calculations (meep is a _linear solver_, useful for analyzing the response of a system given some specific current-source input). (It's also a nonlinear solver, and even has a mode solver wrapped around it using a shift-invert scheme... but that's probably overkill and not really worth it... as described later). 

In particular, for this problem, you probably want a variable meshing mode solver that works well with metals. So not MPB. 

Maybe check out [femwell](https://helgegehring.github.io/femwell/).

In essence, you would solve for the modal fields and from those fields, compute the characteristic impedance etc.
